lib/characteristic.dart

```diff
@@ -0,0 +1,44 @@
+part of flutter_ble_lib;
+
+abstract class _CharacteristicMetadata {
+  static const String uuid = "characteristicUuid";
+  static const String id = "id";
+  static const String isReadable = "isReadable";
+  static const String isWritableWithResponse = "isWritableWithResponse";
+  static const String isWritableWithoutResponse = "isWritableWithoutResponse";
+  static const String isNotifiable = "isNotifiable";
+  static const String isIndicatable = "isIndicatable";
+  static const String value = "value";
+}
+
+class Characteristic {
+  Service service;
+  String uuid;
+  int _id;
+  bool isReadable;
+  bool isWritableWithResponse;
+  bool isWritableWithoutResponse;
+  bool isNotifiable;
+  bool isIndicatable;
+
+  Characteristic(this.uuid, this.service, this._id,
+      {this.isReadable,
+      this.isWritableWithResponse,
+      this.isWritableWithoutResponse,
+      this.isNotifiable,
+      this.isIndicatable});
+
+  Characteristic.fromJson(Map<String, dynamic> jsonObject, Service service)
+      : service = service,
+        _id = jsonObject[_CharacteristicMetadata.id],
+        uuid = jsonObject[_CharacteristicMetadata.uuid],
+        isReadable = jsonObject[_CharacteristicMetadata.isReadable] as bool,
```

Comment:
Casting is a left-over from a silly bug, I'm deleting it. I think it would be better to get a null here and know that there's something wrong on the native side, since all of those fields should be passed over.